web_ui/frontend/app/(login)/login/page.tsx

```diff
@@ -78,86 +76,119 @@ export default function Home() {
                 }
             }
 
-        } catch {
+        } catch (e) {
+            console.error(e)
             setLoading(false)
             setError("Could not connect to server")
         }
     }
 
     function onSubmit(e: React.FormEvent<HTMLFormElement>) {
         e.preventDefault()
-
         submit(password)
     }
 
+    const LoginComponent = (
+        <form onSubmit={onSubmit} action="#">
+            <Box display={"flex"} justifyContent={"center"}>
+                <PasswordInput
+                    FormControlProps={{
+                        sx: {width: "50%"},
+                    }}
+                    TextFieldProps={{
+                        InputProps: {
+                            sx: {width: "50%"},
+                            onChange: (e) => {
+                                setPassword(e.target.value)
+                                setError(undefined)
+                            }
+                        }
+                    }}
+                />
+            </Box>
+            <Box display={"flex"} flexDirection={"column"}>
+                <Grow in={error !== undefined}>
+                    <Typography
+                        textAlign={"center"}
+                        variant={"subtitle2"}
+                        color={"error.main"}
+                        mb={1}
+                    >
+                        {error}
+                    </Typography>
+                </Grow>
+                <LoadingButton
+                    variant="outlined"
+                    sx={{margin: "auto"}}
+                    color={"primary"}
+                    type={"submit"}
+                    loading={loading}
+                >
+                    <span>Login</span>
+                </LoadingButton>
+            </Box>
+        </form>
+    )
+
+    if(enabledServers && (enabledServers.includes("registry") || enabledServers.includes("origin") || enabledServers.includes("cache"))) {
```

Comment:
Sorry I was misreading the code. I was meant to suggest that admin password login should be available to all server UIs, but the **OAuth** should be available for registry, origin, and cache. I'll let you decide what to push in this condition here